src/analysis/analysis-factory.js

```diff
@@ -23,6 +23,8 @@ var AnalysisFactory = function (opts) {
  * may have one or more "source" params pointing to another node. If a node had been created
  * already, this method updates the attributes of the existing node. New nodes are added to
  * the collection of analyses that has been injected.
+ * 
+ * TODO: document what's the analysis definition
```

Comment:
Should we?

Reply:
I missed it for the specs. Depends on what the public API will be. If we ask the users to provide an analysis definition we should.